### ISSUE TITLE
Colon on a line by itself

### ISSUE THREAD

**Issue by paul-j-lucas** [doxygen/doxygen#9309]
**Describe the bug**
With source like this:

```
/**
 * Flag for c_ast_gibberish() to indicate that the declaration is for multiple
 * objects, e.g.:
 *
 *      int *p, *q;
 */
```

it generates HTML where the `:` is put on a line by itself — see screenshot.  If the `e.g.:` is changed to `for example:`, it works as expected. I suspect that Doxygen sees the `.` and thinks it's the end of the sentence and hence the "brief" description.

**Expected behavior**
I expect the `:` to remain next to the `e.g.`.

**Screenshots**
<img width="573" alt="colon" src="https://user-images.githubusercontent.com/7366785/166500993-7bd307de-452d-486c-ab8a-a3e001174bb9.png">

**Version**
1.9.3
macOS 12.3.1
Doxygen via [MacPorts](https://www.macports.org).

**Comment by albert-github** [comment]
Although there are no doxygen settings provided it looks like that the setting of `JAVADOC_AUTOBRIEF` is `YES`.

- Is it on purpose that you have `JAVADOC_AUTOBRIEF = YES`?

When disabling this it works all as you expected.

**Comment by doxygen** [comment]
Another option is to use:

```c
/**
 * Flag for c_ast_gibberish() to indicate that the declaration is for multiple
 * objects, e.g.\ :
 *
 *      int *p, *q;
 */
```

but since ending a brief description with e.g. doesn't really make sense the better solution is to use

```c
/**
 * Flag for c_ast_gibberish() to indicate that the declaration is for multiple
 * objects.
 *
 * Example:
 *      int *p, *q;
 */
```

**Comment by albert-github** [comment]
@doxygen How about extending in `commentscan.l` (around line 951) the rule:
```
<Comment>".,"                           { // . with comma such as "e.g.,"
                                          addOutput(yyscanner,yytext);
                                        }
```
to
```
<Comment>"."[,:;]                           { // . with comma such as "e.g.,"
                                          addOutput(yyscanner,yytext);
                                        }
```

Another point here is that currently the rules regarding the end of brief are all regarding a dot (`.`), but shouldn't this be extended to question marks (`?`) and exclamation marks (`!`) ?

**Comment by paul-j-lucas** [comment]
> Is it on purpose that you have `JAVADOC_AUTOBRIEF = YES`?

Yes, because I hate typing `@brief` all over the place.

> ... since ending a brief description with e.g. doesn't really make sense ...

It makes sense to me.

The proposal for `<Comment>"."[,:;]` seems reasonable to me.

> Another point here is that currently the rules regarding the end of brief are all regarding a dot (`.`), but shouldn't this be extended to question marks (`?`) and exclamation marks (`!`) ?

Maybe.  If you were processing text in general and you needed to find the end of a sentence, then I'd say yes; but this case is specifically for auto-brief.  I can't really see brief descriptions being questions or exclamations (shouting at the reader). OTOH, it's not clear adding `!?` would _hurt_ anything either.

**Comment by doxygen** [comment]
@paul-j-lucas What I meant with "doesn't make sense" is that the brief description is also used as a standalone text at other places, like overviews and tooltips. So for instance the function overview will look like this:
<img width="621" alt="Screenshot 2022-05-04 at 20 14 04" src="https://user-images.githubusercontent.com/1300762/166799681-16dd03ee-12f9-4a84-a0c3-ba45f6ba4d42.png">
The "More..." link may still make it look ok-ish but I hope you get my point.

@albert-github I don't have a problem with the suggested fix.

**Comment by paul-j-lucas** [comment]
Yes, I understand. Personally, I think it's fine.

**Comment by albert-github** [comment]
Regarding the problem with the colon I've just pushed a proposed patch, pull request #9311

**Comment by albert-github** [comment]
Code of pull request #9211 has been integrated in master on GitHub.

The problem regarding the `?` And `!` will be handled in another pull request.

**Comment by albert-github** [comment]
Regarding the recognition of the `?` and `!` as brief terminating characters I've just pushed a proposed patch, pull request #93
Also fixing:
- the fact that `\.` was not handled properly as non-brief terminating command (issue #6205)
- handling of a question mark  in an HTML Help index entry

**Comment by albert-github** [comment]
Code of #9314 has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
